envoy/config/filter/http/jwt_auth/v2/config.proto

```diff
@@ -0,0 +1,168 @@
+  // `here <https://tools.ietf.org/html/rfc7519#section-4.1.1>`_.
+  // Usually a URL or an email address.
+  //
+  // Example: https://securetoken.google.com
+  // Example: 1234567-compute@developer.gserviceaccount.com
+  //
+  string issuer = 1;
+
+  // The list of JWT `audiences
+  // <https://tools.ietf.org/html/rfc7519#section-4.1.3>`_. that are allowed to
+  // access. A JWT containing any of these audiences will be accepted.
+  //
+  // Example::
+  //
+  //     audiences:
+  //     - bookstore_android.apps.googleusercontent.com
+  //       bookstore_web.apps.googleusercontent.com
+  //
+  repeated string audiences = 2;
+
+  // URL of the provider's public key set to validate signature of the
+  // JWT. See `OpenID
+  // Discovery
+  // <https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata>`_
+  //
+  // Example: https://www.googleapis.com/oauth2/v1/certs
+  string jwks_uri = 3;
+
+  // If false, remove the JWT and do not forward to the backend.
+  bool forward_jwt = 4;
+
+  // Duration after which the cached public key should be expired. The
+  // system wide default is applied if no duration is explicitly
+  // specified.
+  google.protobuf.Duration public_key_cache_duration = 5;
+
+  // Two fields below define where to extract the JWT from an HTTP request.
+  //
+  // If no explicit location is specified the following default
+  // locations are tried in order:
+  //
+  // 1. The Authorization header using the Bearer schema.
+  //    See `here <https://tools.ietf.org/html/rfc6750#section-2.1>`_
+  //    Example::
+  //
+  //       Authorization: Bearer <token>.
+  //
+  // 2. `access_token` query parameter.
+  //    See `here <https://tools.ietf.org/html/rfc6750#section-2.3>`_
+  //
+  //
+  // Only the first JWT token is verified per request if multiple tokens are
+  // specified.
+  // If both jwt_headers and jwt_params are specified, jwt_headers will be
+  // checked first.
+  //
+  // JWT is sent in a request header. `jwt_headers` represents the
+  // header names. Their format is::
+  //
+  //    header_name: <JWT>
+  //
+  // For example, if config is::
+  //
+  //   jwt_headers:
+  //      x-goog-iap-jwt-assertion
+  //
+  // The header format will be::
+  //
+  //   x-goog-iap-jwt-assertion: <JWT>.
+  //
+  repeated string jwt_headers = 6;
+
+  // JWT is sent in a query parameter. `jwt_params` represents the
+  // query parameter names.
+  //
+  // For example, if config is::
+  //
+  //   jwt_params:
+  //      jwt_token
+  //
+  // The query parameter is::
+  //
+  //    /path?jwt_token=<JWT>
+  //
+  repeated string jwt_params = 8;
+
+  // This field is specific for Envoy proxy implementation.
+  // It is the cluster name in the Envoy "cluster_manager" config section.
+  // In order for Envoy to call "jwks_uri", its host has to be specified
+  // as a "cluster" in the config for each jwks_uri.
+  string jwks_uri_cluster = 7;
+}
+
+// This is the Envoy filter config for JSON Web Token authentication.
```

Comment:
For better cohesion, consider decoupling Envoy filter config from JWT authentication config.

Reply:
The filer is to do JWT authentication,  so its config will be the JWT authentication config.